pkg/logs/processor/processor.go

```diff
@@ -241,8 +247,14 @@ func (p *Processor) processMessage(msg *message.Message) {
 			return
 		}
 
+		p.utilization.Stop()
 		p.outputChan <- msg
+		p.pipelineMonitor.ReportComponentIngress(msg, "strategy")
+	} else {
+		p.utilization.Stop()
 	}
+	p.pipelineMonitor.ReportComponentEgress(msg, "processor")
```

Comment:
Should this and `.Stop` calls probably need to happen in early returns on error as well?

Reply:
I see you added .Stop(), but what about the `ReportComponentEgress`? If it is not called on all codepaths that forward or drop the message, then `utilization` just keeps growing.

Reply:
True. Ill defer `pipelineMonitor.ReportComponentEgress(msg, "processor")` in this case. 
These errors basically never happen (unless there is an underlying logic error) but it should be correct regardless.